R/add.R

```diff
@@ -301,20 +263,80 @@ m_add_line <- function(
 #'       m_style_sphere(scale = 0.3)
 #'     )
 #'   ) %>%
-#'   m_add_lines(line_specs = line_specs)
+#'   m_add_lines(
+#'     starts = list(
+#'       m_sel(resi = 1, chain = "A"),
+#'       m_sel(resi = 1, chain = "A")),
+#'     ends = list(
+#'       m_sel(resi = 10, chain = "A"),
+#'       m_sel(resi = 10, chain = "B")
+#'     ),
+#'     dashed = TRUE
+#'   )
 m_add_lines <- function(
                         id,
-                        line_specs) {
-  if (missing(line_specs) || is.null(line_specs)) {
-    stop("`line_specs` should be passed in.")
+                        starts,
+                        ends,
+                        dashed = TRUE,
+                        color = "black",
+                        opacity = 1,
+                        hidden = FALSE
+                        ) {
+  if (missing(starts) | missing(ends)) {
+    stop("At least 1 start and 1 end must be passed in.")
+  }
+
+  if (methods::is(starts)[1] == "AtomSelectionSpec") {
+    starts <- list(starts)
   }
+  if (methods::is(ends)[1] == "AtomSelectionSpec") {
+    ends <- list(ends)
+  }
+
+  line_specs <- .m_multi_spec(
+    starts = starts,
+    ends = ends
+  )
+
+  if (length(starts) != length(ends)) {
+    stop(paste("ERROR length(starts) must equal length(ends)."))
+  }
+
+  test_length <- function(option) {
+    if (length(option) != length(starts)) {
+      if (length(option) != 1) {
+        stop(paste(deparse(substitute(option)), "options must of length 1 or",
+                   "equal to the number of line starts & stops."))
+      }
+    }
+  }
+
+  test_length(dashed)
+  test_length(color)
+  test_length(opacity)
+  test_length(hidden)
```

Comment:
It wouldn't be a good idea about define a function in another function. Would be better to move it to utils and add tests.